### PR TITLE
Be selective about when to use foreign keys for mapping properties

### PR DESCRIPTION
**Pull request by snurra**

### CODE REVIEW

File: backbone-despotism.js [hunk 1]
```diff
@@ -42,6 +45,15 @@
   }
 } (this, function(exports, Backbone, _) {
   Backbone.StrictModel = Backbone.Model.extend({
+    constructor: function(attributes, options) {
+      // Make sure that the options object when initializing a new instance of
+      // the StrictModel always has { initialize: true }
+      attributes = attributes || (attributes = {});
```
Comment (snurra) [inline backbone-despotism.js]: Why not just
```JavaScript
attributes = attributes || {};
```
?

File: backbone-despotism.js [hunk 2]
```diff
@@ -42,6 +45,15 @@
...
+      options || (options = {});
```
Comment (snurra) [inline backbone-despotism.js]: And isn't the same style here just a liiiiittle bit clearer? 😜  (I like when it's obvious when you're declaring something.)

File: backbone-despotism.js [hunk 3]
```diff
@@ -73,11 +85,18 @@
     }
   });
   
-  function mapAttributes (json, props, defaults) {
+  function mapAttributes (json, props, defaults, options) {
+    var self = this;
```
Comment (snurra) [inline backbone-despotism.js]: This poor `self` is alone and unused.

File: backbone-despotism.js [hunk 4]
```diff
@@ -73,11 +85,18 @@
...
-      // Foreign key (if it exists) should only be used if the key does not exist in the json or it has the default value
-      var foreignKey = (definition.foreignKey && (typeof json[key] === "undefined" || (defaults && json[key] === defaults[key] && typeof json[definition.foreignKey] !== "undefined"))) ? definition.foreignKey : key;
+      var foreignKey = key;
+      // Foreign keys should only be used on initialization of the model, or if
+      // specifically requested
+      if (options.useForeignKeys || options.initialize || options.reset) {
+        // Foreign key (if it exists) should only be used if the key does not exist in the json or it has the default value
+        foreignKey = (definition.foreignKey && (typeof json[key] === "undefined" || (defaults && json[key] === defaults[key] && typeof json[definition.foreignKey] !== "undefined"))) ? definition.foreignKey : key;
+      }
       var type = definition.type || definition;
```
Comment (snurra) [inline backbone-despotism.js]: Perhaps move the declaration to the top of the function?

File: backbone-despotism.js (hunk 1 above)
Comment (jsahlen) [reply backbone-despotism.js]: Yes.

File: backbone-despotism.js (hunk 2 above)
Comment (jsahlen) [reply backbone-despotism.js]: Yes.

File: backbone-despotism.js (hunk 3 above)
Comment (jsahlen) [reply backbone-despotism.js]: Yes.

File: backbone-despotism.js (hunk 4 above)
Comment (jsahlen) [reply backbone-despotism.js]: Yes.

**Comment by snurra** [comment]
image

**Comment by jsahlen** [comment]
Yes.

**Pull request merged**